src/functional/ffront/past_to_itir.py

```diff
@@ -196,21 +198,17 @@ def _visit_stencil_call_out_arg(
             fun=itir.SymRef(id="domain"), args=domain_args
         )
 
-    def visit_Constant(
-        self, node: past.Constant, **kwargs
-    ) -> Union[itir.IntLiteral, itir.FloatLiteral, itir.BoolLiteral]:
+    def visit_Constant(self, node: past.Constant, **kwargs) -> Union[itir.Literal]:
         if isinstance(node.type, common_types.ScalarType) and node.type.shape is None:
             match node.type.kind:
-                case common_types.ScalarKind.INT32 | common_types.ScalarKind.INT64:
-                    return itir.IntLiteral(value=node.value)
-                case common_types.ScalarKind.FLOAT32 | common_types.ScalarKind.FLOAT64:
-                    return itir.FloatLiteral(value=node.value)
-                case common_types.ScalarKind.BOOL:
-                    return itir.BoolLiteral(value=node.value)
-                case _:
+                case common_types.ScalarKind.STRING:
                     raise NotImplementedError(
                         f"Scalars of kind {node.type.kind} not supported currently."
                     )
+            typename = node.type.kind.name.lower()
+            if typename.startswith("int"):
+                typename = "int"
```

Comment:
```suggestion
            if typename.startswith("int"):
                typename = "int"
            elif typename.startswith("float"):
                typename = "float"
            assert typename in ["bool", "int", "float"]
```

Reply:
There is a reason behind reverting ints in the program function to unspecified ints (up to the backend which int to choose for loop variables, domain sizes etc). There seems to be no similar reasoning for floats?